test/acceptance/hooks/mouse-position.acceptance.test.js

```diff
@@ -0,0 +1,42 @@
+import globals from '../globals'
+import Storybook from '../storybook'
+
+const storybook = new Storybook()
+
+fixture('Mouse Position Hook')
+  .page(
+    globals.url +
+      '/?selectedKind=Mouse Position&selectedStory=Default&full=0&addons=1&stories=1&panelRight=1&addonPanel=REACT_STORYBOOK%2Freadme%2Fpanel'
+  )
+  .beforeEach((t) => t.switchToIframe(storybook.iframe))
+  .afterEach((t) => t.switchToMainWindow())
+
+test('The mouse position demo is rendered', async (t) => {
+  const { title } = storybook.hooks.mousePosition
+  await t.expect(title.textContent).contains('Mouse Position Demo')
+})
+
+test('The mouse position defaults to 0,0', async (t) => {
+  const { description } = storybook.hooks.mousePosition
+  await t.expect(description.textContent).contains('X: 0px, Y: 0px')
+})
+
+test('The mouse position updates when the mouse moves', async (t) => {
+  const { description, html } = storybook.hooks.mousePosition
+
+  await t.hover(html)
```

Comment:
I'm not sure I understand this test. I think it implicitly assumes that doing this ^ operation moves the mouse in a specific place, which you assert below. I think it's worth specifying what is the expectation here because it's not intuitive.

Reply:
I'll add a comment because it's not very intuitive. Basically, testcafe doesn't have a `mousemove` action. The only way to fire a mouseMove event is to hover over an element. By default, the mouse will hover over the center of the element. This test calculates where the center of the element would be and asserts that the mousePosition reported by the hook is the true position (center of the element). The center couldn't be hard coded because it's slightly different for each browser

Reply:
makes sense, worth a comment though